### DISCUSSION TITLE
Warnings in OpenMc Simulation

### DISCUSSION THREAD

**Discussion by AnthonyB08** [svalinn/DAGMC#854] [Q&A] [answered]
Hi all, 

My name is Anthony and I am excited to start my journey in computational neutronics. I am very new to this software ( both OpenMC and DAGMC) but I have a nuclear background and have read most of the documentation. That being said I am trying to solve a simple neutron transport problem being a spherical ball filled with plutonium (BC's vacuum). I have created the geometry and mesh using Cubit  Coreform with the mesh being tetmesh scheme as seen below: 
![image](https://user-images.githubusercontent.com/55299169/217069925-39e1c962-83d9-443e-ae17-a975d22ced7a.png)

I export the file as a DAGMC .h5m file extension and Import it correctly as seen below:
# Create DAGMC universes out of h5m files
NeutronBall= openmc.DAGMCUniverse(filename=mesh_file)
print(NeutronBall.material_names)
print(NeutronBall.bounding_region)
geo = openmc.Geometry(root=NeutronBall)
geo.export_to_xml()

Output: 
![image](https://user-images.githubusercontent.com/55299169/217070396-38fcce7f-c1b0-4f7d-80eb-82a179a843ba.png)

Next, I create a mesh filer so I can determine the neutron flux spatially as such : 
![image](https://user-images.githubusercontent.com/55299169/217070562-57a7d658-cd0e-4452-8b1d-c41b2fcb45b2.png)

I, Then, run the simulation. During this time openmc returns these warnings:
Warning #1 : 
![image](https://user-images.githubusercontent.com/55299169/217070779-b845a4fb-3aac-4d3b-ad3e-f9468fb032eb.png)
Warning#2:
![image](https://user-images.githubusercontent.com/55299169/217070937-46497d21-23c1-4d1d-abf3-0e9cccd59b6b.png)

Then when I go to look at the tallie.out file I see no data was stored

What does the community recommend/advise? 

Thank you very much! :)

**Comment by AnthonyB08** [comment, marked as answer]
First warning is removed if i add  .bounded_universe to the openc.DAGMCUniverse(filename="NeutronBall.h5m")

**Reply by shimwell** [reply]
Yes this adds the missing surface with a vacuum boundary type to encompass the model automatically. Good find

```python
openc.DAGMCUniverse(filename="NeutronBall.h5m").bounded_universe
```

**Comment by AnthonyB08** [comment]
Update: I wanted to see if I was missing any packages so I ran a known-working solution, i.e https://github.com/openmc-dev/openmc-notebooks/blob/main/unstructured-mesh-part-ii.ipynb. When I run this notebook all works well and converges, however when I go o paraview and import the .vtk file it shows no neutron flux. Note that I did not change anything in this note book. 

![image](https://user-images.githubusercontent.com/55299169/217297308-10003629-231d-4bb2-83d8-5e24bed68f4b.png)

**Reply by shimwell** [reply]
Is there any chance you can attach the vtk file and I can take a look over here

**Reply by AnthonyB08** [reply]
I cannot attach the file directly to GitHub . what would be the best way to send the files?

**Comment by AnthonyB08** [comment]
I am wondering if the second warning has anything to do with how MOAB is installed, as well as any of the installed packages. 

I also noticed that when I export the CUBIT mesh to dagmc , through the plugin, the mesh is altered when doing a mbconvert h5m to vtx.

**Reply by shimwell** [reply]
Perhaps a zip file is [accepted by github](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files)

**Comment by pshriwise** [comment]
Thanks for bringing this to our attention @AnthonyB08! I'll pull that notebook down and run it as well today to confirm. Hopefully I'll be able to get back to you soon.

**Comment by pshriwise** [comment]
Hi @AnthonyB08,

I re-ran that notebook and the VTK file it produced did contain the expected datasets when I opened it in Paraview. 

Are you sure all of those values are truly zero? The scale indicates that the values have a max of .0022. I noted the scale of your flux was linear in the image above. It changes many orders of magnitude in this problem, so you might try it with log scaling instead.

![image](https://user-images.githubusercontent.com/4563941/217841133-e2542fd4-7455-4372-b102-30fbc3b61fde.png)

**Reply by shimwell** [reply]
I get the same as you Patrick
